### DISCUSSION TITLE
Using ephemerals to select variants

### DISCUSSION THREAD

**Discussion by Anti-Distinctlyminty** [AcademySoftwareFoundation/rez#1546] [Q&A]
I'm attempting to setup a package for ACES configurations and have run into a issue where ephemerals in variants do not act like I expected.

See the following packages.py

```python
name = "aces"

description = "OCIO ACES configuration"

version = "1.3.0"

build_command = False

requires = ["ocio_configs-2+"]

variants = [
    [".colorpaces-1.0.0", ".ocio-2.1"],
    [".colorspaces-2.1.0", ".ocio-2.3"],
]

def commands():
    import os
    
    env.OCIO = os.path.normpath(
        "{ocio_configs_root}/OpenColorIO-{ocio_configs_version}/src/OpenColorIO/builtinconfigs/configs/cg-config-v{colorpsaces_version}_aces-v{aces_major_minor}_ocio-v{ocio_major_minor}.ocio".format(
            ocio_configs_root=resolve.ocio_configs.root,
            ocio_configs_version=resolve.ocio_configs.version,
            colorpsaces_version=ephemerals.get_range("colorspaces"),   
            aces_major_minor=f"{this.version[0]}.{this.version[1]}",
            ocio_major_minor=ephemerals.get_range("ocio"),
        )
    )

    if not os.path.exists(str(env.OCIO)):
        stop(f"Specified .ocio does not exist: {env.OCIO}")
```
The ocio_configs package are the configs avialable from [the OCIO Github](https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/releases)
My intent was to use ephemerals and variants as a kind of switch to allow the correct .ocio file to be set.
e.g. if the user requests `rez-env aces rez-env aces-1.3 .colorspaces-1`, then _I thought_ it would automatically pick up the variant involving `.colorspaces-1.0.0`, resulting in choosing the 
`ocio_configs\2.1.0\OpenColorIO-2.1.0\src\OpenColorIO\builtinconfigs\configs\cg-config-v1.0.0_aces-v1.3_ocio-v2.1.ocio` file. However, this is not the case as the ephemeral is just passed as is and has no effect on the variant chosen. So the .ocio file is actually 
`ocio_configs\2.1.0\OpenColorIO-2.1.0\src\OpenColorIO\builtinconfigs\configs\cg-config-v1_aces-v1.3_ocio-v2.1.ocio`

So I'm guessing that the ephemerals in combination with variants do not act this way? If not, is there another way I can achieve the same thing?

**Comment by JeanChristopheMorinPerso** [comment]
Interesting, my intuition tells me that what you expect is what I would expect too. But I've never used ephemerals before so maybe I'm missing something.

**Reply by Anti-Distinctlyminty** [reply]
We were indeed correct. There was a typo in the variants which I somehow missed `colorpaces`. It now works as expected.
